ios/browser/api/omnibox/autocomplete_classifier.mm

```diff
@@ -0,0 +1,125 @@
+// Copyright (c) 2024 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#include "brave/ios/browser/api/omnibox/autocomplete_classifier.h"
+
+#include "base/strings/sys_string_conversions.h"
+#include "base/strings/utf_string_conversions.h"
+#include "components/omnibox/browser/autocomplete_classifier.h"
+#include "components/omnibox/browser/autocomplete_match.h"
+#include "components/prefs/pref_service.h"
+#include "ios/chrome/browser/autocomplete/model/autocomplete_classifier_factory.h"
+#include "ios/chrome/browser/shared/model/application_context/application_context.h"
+#include "ios/chrome/browser/shared/model/browser_state/chrome_browser_state.h"
+#include "ios/chrome/browser/shared/model/browser_state/chrome_browser_state_manager.h"
+#include "ios/chrome/browser/shared/model/prefs/pref_names.h"
+#include "net/base/apple/url_conversions.h"
+
+namespace brave {
+AutocompleteMatch::Type MatchTypeFromBraveType(
+    BraveIOSAutocompleteMatchType type) {
+  switch (type) {
+    case BraveIOSAutocompleteMatchTypeUrlWhatYouTyped:
```

Comment:
We need to look at some better way to do this in general. Probably something like what android does to auto generate java enums? Not for this PR though